src/promptflow/tests/sdk_cli_test/e2etests/test_flow_run.py

```diff
@@ -1576,3 +1576,33 @@ def test_run_with_non_provided_connection_override_list_annotation(self, pf, loc
             "inputs.line_number": [0],
             "outputs.output": [{"connection": "Custom", "key": "API_BASE"}],
         }
+
+    def test_run_with_init(self, pf):
+        def assert_func(details_dict):
+            return details_dict["outputs.func_input"] == [
+                "func_input",
+                "func_input",
+                "func_input",
+                "func_input",
+            ] and details_dict["outputs.obj_input"] == ["val", "val", "val", "val"]
+
+        flow_path = Path(f"{EAGER_FLOWS_DIR}/basic_callable_class")
+        run = pf.run(
+            flow=flow_path, data=f"{EAGER_FLOWS_DIR}/basic_callable_class/inputs.jsonl", init={"obj_input": "val"}
+        )
```

Comment:
the init param `val` is the same with yaml, how to know which one is taking effect?

Reply:
and which one is at higher priority when both are present?

Reply:
Do you mean the `run.yaml` of `flow.dag.yaml`? 
We don't support specify both `run.yaml` and `init`.
For `flow.dag.yaml`, we currently don't support write `init` in it.

Reply:
After we support init in flow.dag.yaml, I think the priority will be:
run param > flow.dag.yaml > code.

Reply:
I see, they are two different test cases.